extension/src/features/approve_pasted_psbt.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{
+    components::{
+        select::{Select, SelectItem},
+        textarea::TextArea,
+    },
+    context::UserContext,
+    utils::{helpers::get_clipboard, storage::LocalStorage},
+};
+use anyhow::anyhow;
+use anyhow::Result;
+use signer::{
+    psbt_details::PSBTDetails,
+    signer::decode_psbt_and_sign,
+    storage::{SettingsStorage, UserStorage},
+};
+use std::str::FromStr;
+use yew::prelude::*;
+use yew_router::prelude::use_navigator;
+
+#[function_component(ApprovePastedPSBT)]
+pub fn approve_pasted_psbt() -> Html {
+    let password = use_context::<UserContext>()
+        .unwrap()
+        .password
+        .clone()
+        .unwrap_or_default();
+    let navigator = use_navigator().unwrap();
+    let storage = UserStorage::read(LocalStorage::default());
+    let default_wallet = storage.get_default_wallet();
+    let selected_wallet = use_state(|| default_wallet);
+    let error = use_state(String::default);
+    let selected_wallet_value = (*selected_wallet).clone();
+    let error_value = (*error).clone();
+    let disabled = password.is_empty();
+    let psbt = use_state(String::default);
+    let psbt_value = (*psbt).clone();
+    let signed_psbt = use_state(String::default);
+    let signed_psbt_value = (*signed_psbt).clone();
+
+    let onchange_psbt = {
+        let psbt = psbt.clone();
+        Callback::from(move |value: Result<String>| {
+            let _ = value.map(|v| psbt.set(v));
+        })
+    };
+
+    let onclick_save = {
+        let selected_wallet_value = selected_wallet_value.clone();
+        let psbt = psbt.clone();
+        Callback::from(move |_: MouseEvent| {
+            if password.is_empty() {
+                return;
+            }
```

Comment:
does the user know that the password is empty? how does yew process this? asks the user for a password again?

Reply:
Button is disabled